kodi_addon_checker/check_addon.py

```diff
@@ -384,3 +389,31 @@ def _check_dependencies(report: Report, addon_path, repo_addons):
             if LooseVersion(available_version) < LooseVersion(required_version) and (required_addon not in ignore):
                 report.add(Record(PROBLEM, "Version mismatch for addon %s. Required: %s, Available: %s "
                                   % (required_addon, required_version, available_version)))
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(report: Report, addon_path, all_repo_addons):
+    """Check if addon submitted already exists or not"""
+    addon_xml = os.path.join(addon_path, "addon.xml")
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    branches = ['gotham', 'helix', 'isengard', 'jarvis', 'krypton', 'leia']
+
+    for branch in branches:
```

Comment:
`for branch in branches`

Reply:
sorry I don't understand this ?

Reply:
sorry wrong variable
`for branch in all_repo_addons:` and get rid of the now unused variable branches
defining branches 2 times is error prone

Reply:
ok but then for that I'll have to reverse the order of branches in the `branches` list so that gotham is the first branch and leia is the last one because then if krypton will be the last one added to the dictionary then we'll have different output

if `for branch in branches:` then
```python
INFO: Checking add-on plugin.video.twitch
INFO: Created by A Talented Community
INFO: Addon id matches folder name
**INFO: plugin.video.twitch addon also exists in  gotham branch but with version: 1.4.1**
INFO: Image icon exists
INFO: icon dimensions are fine 512x512
INFO: Image fanart exists
INFO: fanart dimensions are fine 3840x2160
WARN: Complex entry point. Check: resources/lib/routes.py | Counted lines: 1286 | Lines allowed: 80
WARN: Complex entry point. Check: resources/lib/service.py | Counted lines: 155 | Lines allowed: 80
WARN: We found no problems and 2 warnings, please check the logfile.
```
--------------------------------------------------------------------------
if `for branch in all_repo_addons:` then
```python
INFO: Checking add-on plugin.video.twitch
INFO: Created by A Talented Community
INFO: Addon id matches folder name
**ERROR: plugin.video.twitch addon already exist with version: 2.2.2 in krypton branch**
INFO: Image icon exists
INFO: icon dimensions are fine 512x512
INFO: Image fanart exists
INFO: fanart dimensions are fine 3840x2160
WARN: Complex entry point. Check: resources/lib/routes.py | Counted lines: 1286 | Lines allowed: 80
WARN: Complex entry point. Check: resources/lib/service.py | Counted lines: 155 | Lines allowed: 80
ERROR: We found 1 problems and 2 warnings, please check the logfile.
```
Notice the difference in the `Error` and `Info` message